### PR TITLE
Using conda-forge's tzdata on all platforms

### PR DESCRIPTION
**Pull request by SylvainCorlay**
With this PR

 - tzdata becomes a dependency of `howardhinnant_date` on all platforms.
 - we compile `howardhinnant_date` with `USE_SYSTEM_TZ_DB` set to `ON` on all platforms.
 - the `discover_tz_dir` is patched to return the tzdata directory under the conda prefix instead of that of the system.
 - the cmake file is patched to allow using this compilation option on windows.
 
 The patches apply to windows largely come from https://github.com/HowardHinnant/date/pull/611 - except that we use `dirent` to list files and directories on Windows like it is already done on Linux, making the patch smaller.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@jjerphan I have tested that this works in the sparrow tests on both Windows and Linux.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,19 +10,24 @@ source:
...
 
 build:
-  number: 3
+  number: 4
   skip: true  # [win and vc<14]
 
 requirements:
   build:
     - {{ compiler('cxx') }}
-    - {{ stdlib('c') }}
```
Comment (jjerphan) [inline recipe/meta.yaml:21]: Is there a reason for removing this?

**Review by jjerphan — APPROVED** [review]
:pray:

File: recipe/meta.yaml (hunk 1 above)
Comment (SylvainCorlay) [reply recipe/meta.yaml:21]: It does not work when running conda-build from osx-arm platform. Happy to re-add.

Comment (JohanMabille) [inline recipe/meta.yaml:21]: This should be added back, but let's do it in a dedicated PR.

**Review by JohanMabille — APPROVED** [review]
